src/app/components/auth/signup/RegisterForm.tsx

```diff
@@ -0,0 +1,370 @@
+'use client';
+
+import { formSchema } from '@/lib/yup/signupValidationSchema';
+import { yupResolver } from '@hookform/resolvers/yup';
+import Link from 'next/link';
+import { useForm } from 'react-hook-form';
+import { ErrorMessage } from '../../shared/ErrorMessage';
+import React, { ChangeEvent, useContext, useState } from 'react';
+import { PasswordStrengthBar } from './PasswordStrengthBar';
+import { useRouter } from 'next/navigation';
+import Image from 'next/image';
+import {
+  SignUpErrorCodes,
+  formValidationErrors,
+  locale,
+} from '@/locales/locale';
+import { localeContext } from '@/locales/localeProvider';
+
+interface FormData {
+  name: string;
+  email: string;
+  password: string;
+  passwordConfirmation: string;
+  terms?: boolean | undefined;
+}
+
+interface ErrorResponse {
+  message: string;
+  errorCode: SignUpErrorCodes;
+}
+
+export function RegisterForm() {
+  const router = useRouter();
+  const { state } = useContext(localeContext);
+  const currentLang = state.currentLocale.id;
+
+  const {
+    alreadyHaveAccount,
+    signinLink,
+    termsLink,
+    signupBtn,
+    nameLabel,
+    namePlaceholder,
+    emailLabel,
+    passwordLabel,
+    confirmPasswordLabel,
+    termsLabel,
+  } = locale[currentLang];
+
+  const [password, setPassword] = useState('');
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({
+    mode: 'onChange',
+    resolver: yupResolver(formSchema),
+  });
+
+  const {
+    onChange: onNameChange,
+    onBlur: onNameBlur,
+    name: nameName,
+    ref: nameRef,
+  } = register('name');
+
+  const {
+    onChange: onEmailChange,
+    onBlur: onEmailBlur,
+    name: emailName,
+    ref: emailRef,
+  } = register('email');
+
+  const {
+    onChange: onPasswordChange,
+    onBlur: onPasswordBlur,
+    name: passwordName,
+    ref: passwordRef,
+  } = register('password');
```

Comment:
please consider using rest/spread operator here, you can look at documentation of itself(react-hook-form)

Reply:
@iamnkt
Please take this task.

Reply:
fixed in https://github.com/thirdmadman/graphiql-app/pull/70/commits/fb2f43319a0d35729536b5b879cc0d192d0f6172